packages/server-wallet/src/wallet/__test__/fixtures/test-channel.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  Address,
+  BN,
+  calculateChannelId,
+  ChannelConstants,
+  makeAddress,
+  makePrivateKey,
+  Objective,
+  Outcome,
+  Participant,
+  PrivateKey,
+  serializeState,
+  SignedStateWithHash,
+  simpleEthAllocation,
+  State,
+} from '@statechannels/wallet-core';
+import {ETH_ASSET_HOLDER_ADDRESS} from '@statechannels/wallet-core/lib/src/config';
+import {SignedState as WireState, Payload} from '@statechannels/wire-format';
+
+import {SigningWallet} from '../../../models/signing-wallet';
+import {WALLET_VERSION} from '../../../version';
+
+import {alice, bob} from './participants';
+import {alice as aliceWallet, bob as bobWallet} from './signing-wallets';
+import {stateWithHashSignedBy} from './states';
+
+/** A two-party channel between Alice and Bob, with state history. For testing purposes. */
+export class TestChannel {
+  public participantA: Participant = alice();
+  public participantB: Participant = bob();
+
+  public signingWalletA: SigningWallet = aliceWallet();
+  public signingWalletB: SigningWallet = bobWallet();
+  public startBals: [number, number];
+  public channelNonce = 5;
+
+  public get participants(): Participant[] {
+    return [this.participantA, this.participantB];
+  }
+  public get signingWallets(): SigningWallet[] {
+    return [this.signingWalletA, this.signingWalletB];
+  }
+
+  public static create(aBal: number, bBal: number): TestChannel {
+    return new TestChannel(aBal, bBal);
+  }
+
+  private constructor(aBal: number, bBal: number) {
+    this.startBals = [aBal, bBal];
+  }
+
+  /**
+   * Gives the nth state in the history, signed by the correct participant
+   *
+   * @param n - the index of the state in the history
+   *
+   * Note - in cases where participants double-sign the same states, n might _not_
+   * be the turnNum
```

Comment:
Not sure I understand this Note?

Reply:
What I mean is if we're following the "A signs turnNum 0, B signs turnNum 0" school of the prefundsetup, then `state(0)` would be turnNum 0 with A's signature, and `state(1)` would be turnNum 0 with B's signature (i.e. what B sends back to A).

Reply:
ok it makes sense to me now